docs/user-guides/federation.md

```diff
@@ -0,0 +1,165 @@
+# Federate Monitoring Stack and Openshift In-Cluster Prometheus
+
+## Architecture / Topology
+
+![Architecture](federation/assets/cmo-obo-federation.png)
+
+This example deploy a MonitoringStack in `federate-cmo` namespace and reads
+only a selected set of metrics from the in-cluster prometheus.
+
+## Steps
+
+Assuming that ObO is installed, follow the steps below to create a
+MonitoringStack that uses the in-cluster prometheus `/federate` endpoint to
+scrape selected metrics.
+
+**NOTE:** All examples used in this guide can be found under the [manifests
+directory](federation/manifests)
+
+### Create a Project to deploy MonitoringStack
+
+
+```yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: federate-cmo
+spec:
+```
+
+
+### Deploy Monitoring Stack
+
+
+Apply the following MonitoringStack 
+
+```yaml
+apiVersion: monitoring.rhobs/v1alpha1
+kind: MonitoringStack
+metadata:
+  name: federate-cmo-ms
+  namespace: federate-cmo
+spec:
+  namespaceSelector:  # 👈 restrict to openshift-monitoring namespace
+    matchLabels:
+      kubernetes.io/metadata.name: openshift-monitoring
+
+  resourceSelector:  # 👈 this is used to select the ServiceMonitor
+    matchLabels:
+      monitoring.rhobs/stack: federate-cmo-ms
+
+  logLevel: info # 👈 debug for verbose logs 
+  retention: 3h
+
+  alertmanagerConfig:
+    disabled: true
+
+  prometheusConfig:
+    replicas: 1
+    enableRemoteWriteReceiver: true
+
+  resources: # 👈 ensure that you provide sufficient amount of resources
+    requests:
+      cpu: 500m
+      memory: 1Gi  
+```
+
+
+### Grand permission to Federate In-Cluster Prometheus
+
+```yaml
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: federate-cmo-ms-view
+  labels:
+    kubernetes.io/part-of: federate-cmo-ms
+    monitoring.rhobs/stack: federate-cmo-ms
+
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cluster-monitoring-view
+subjects:
+- kind: ServiceAccount
+  name: federate-cmo-ms-prometheus  # 👈  ServiceAccount used in the prometheus deployed by ObO
+  namespace: federate-cmo
+```
+
+### Create ServiceMonitor for Federation
```

Comment:
I have a slightly different ServiceMonitor that can be created in the federate-cmo namespace. It should also create only 1 target per Prometheus pod.

```
apiVersion: monitoring.rhobs/v1
kind: ServiceMonitor
metadata:
  name: federate-cmo
  namespace: federate-cmo
spec:
  endpoints:
  - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
    honorLabels: true
    interval: 30s
    params:
      match[]:
      - '{__name__=~"(container|pod):.*",namespace="federate"}'
    path: /federate
    port: web
    relabelings:
    # hard-code the target's address.
    - action: replace
      replacement: prometheus-k8s.openshift-monitoring.svc:9091
      targetLabel: __address__
    - action: labeldrop
      regex: pod|namespace|service|endpoint|container
    - action: replace
      replacement: my-openshift-cluster
      targetLabel: source
    scheme: https
    tlsConfig: # to be continued
      insecureSkipVerify: true
  selector: # use the federate-prometheus service to create a "dummy" target.
    matchLabels:
      app.kubernetes.io/managed-by: observability-operator
      app.kubernetes.io/name: federate-prometheus

```

Reply:
How does this ServiceMonitor use the `/federate` endpoint of incluster prometheus?